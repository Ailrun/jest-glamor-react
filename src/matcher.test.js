import {toMatchSnapshot} from 'jest-snapshot'
import {toMatchSnapshotWithGlamor} from './matcher'

jest.mock('jest-snapshot', () => ({
  toMatchSnapshot: jest.fn(),
}))

test('passes', () => {
  toMatchSnapshot.mockImplementation(() => ({pass: true}))
  const result = toMatchSnapshotWithGlamor('stuff')
  expect(result).toEqual({pass: true})
})

test('formats a message', () => {
  const expected = `
    .css-1otybxc,
    [data-css-1otybxc] {
      padding: 4em;
      background: papayawhip;
    }

    .css-1tnuino,
    [data-css-1tnuino] {
      font-size: 1.5em;
      text-align: center;
      color: palevioletred;
    }

    <section
      className="some-other-class css-1otybxc"
    >
      <h1
        className="changed-class css-1tnuino"
      >
        Hello World, this is my first glamor styled component!
      </h1>
    </section>
  `
  const actual = `
    .css-i77z7r,
    [data-css-i77z7r] {
      padding: 4em;
      background: blue;
    }

    .css-1252hns,
    [data-css-1252hns] {
      font-size: 1.5em;
      text-align: left;
      color: palevioletred;
    }

    <section
      className="some-other-class css-i77z7r"
    >
      <h1
        className="class-changed css-1252hns"
      >
        Hello World, this is my first glamor styled component!
      </h1>
    </section>
  `
  toMatchSnapshot.mockImplementation(() => ({pass: false, expected, actual}))
  const mockRecieved1 = Symbol('mock-received1')
  const mockRecieved2 = Symbol('mock-received2')
  const result = toMatchSnapshotWithGlamor(mockRecieved1, mockRecieved2)
  expect(toMatchSnapshot).toHaveBeenCalledWith(mockRecieved1, mockRecieved2)
  expect(result.pass).toBe(false)
  expect(result.message).toMatchSnapshot()
})

test('formats a message with data- attributes', () => {
  const expected = `
    .css-12gg9yz,
    [data-css-12gg9yz] {
      background-color: rebeccapurple;
      margin: 2px;
    }

    <div
      data-css-12gg9yz=""
    />
  `

  const actual = `
    .css-1rdlmfe,
    [data-css-1rdlmfe] {
      background-color: rebeccapurples;
      margin: 2px;
    }

    <div
      data-css-1rdlmfe=""
    />
  `
  toMatchSnapshot.mockImplementation(() => ({pass: false, expected, actual}))
  const mockRecieved1 = Symbol('mock-received1')
  const mockRecieved2 = Symbol('mock-received2')
  const result = toMatchSnapshotWithGlamor(mockRecieved1, mockRecieved2)
  expect(toMatchSnapshot).toHaveBeenCalledWith(mockRecieved1, mockRecieved2)
  expect(result.pass).toBe(false)
  expect(result.message).toMatchSnapshot()
})
