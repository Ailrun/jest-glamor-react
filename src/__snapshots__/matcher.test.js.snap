// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats a message 1`] = `
"[31m- Expected[39m
[32m+ Received[39m

[90m  [39m
[37m-     .css-1otybxc,[39m
[37m-     [data-css-1otybxc] {[39m
[37m+     .css-i77z7r,[39m
[37m+     [data-css-i77z7r] {[39m
[90m        padding: 4em;[39m
[31m-       background: papayawhip;[39m
[32m+       background: blue;[39m
[90m      }[39m
[90m  [39m
[37m-     .css-1tnuino,[39m
[37m-     [data-css-1tnuino] {[39m
[37m+     .css-1252hns,[39m
[37m+     [data-css-1252hns] {[39m
[90m        font-size: 1.5em;[39m
[31m-       text-align: center;[39m
[32m+       text-align: left;[39m
[90m        color: palevioletred;[39m
[90m      }[39m
[90m  [39m
[90m      <section[39m
[37m-       className=\\"css-1otybxc\\"[39m
[37m+       className=\\"css-i77z7r\\"[39m
[90m      >[39m
[90m        <h1[39m
[37m-         className=\\"css-1tnuino\\"[39m
[37m+         className=\\"css-1252hns\\"[39m
[90m        >[39m
[90m          Hello World, this is my first glamor styled component![39m
[90m        </h1>[39m
[90m      </section>[39m
[90m    [39m"
`;
